<launch>
    <arg name="port" default="/dev/ttyACM0"/>
    <arg name="arduino_baud" default="57600"/>
    <arg name="robot_name" default="biba"/>
    <arg name="arduino_connected" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="launch_arduino_node" default="true"/>

    <group if="$(arg launch_arduino_node)">
        <node pkg="bigbang_eurobot" type="arduino_interface.py" name="arduino_interface" output="screen" respawn="true" respawn_delay="1"> 
            <param name = "debug" value = "$(arg debug)"/>
            <param name = "allow_debug_connect" value = "false"/>
        </node>
    </group>
    <group if="$(arg arduino_connected)">
        <node pkg="rosserial_arduino" type="serial_node.py" name="arduino_serial_node" output="screen" respawn="true" respawn_delay="1">
            <param name="port" value="$(arg port)"/>
            <param name="baud" value="$(arg arduino_baud)"/>
        </node>
    </group>

</launch>